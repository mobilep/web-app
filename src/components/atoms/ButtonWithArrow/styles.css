.ButtonWithArrow {
	font-weight: var(--normal);
}

.ButtonWithArrow span {
	display: inline-block;
	position: relative;
	transition: all .3s ease-out;
	will-change: transform;
}

.ButtonWithArrow:hover span {
	transform: translateX(-13px);
}

.ButtonWithArrow svg {
	width: 16px;
	height: 16px;
	position: absolute;
	right: 0;
	opacity: 0;
	top: 50%;
	transform: translate(0, -50%);
	transition: all .3s ease-out;
	will-change: transform, opacity;
}

.ButtonWithArrow:hover svg {
	opacity: 1;
	transform: translate(26px, -50%);
}
