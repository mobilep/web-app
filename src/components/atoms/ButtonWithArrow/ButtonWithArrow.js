import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { ArrowIcon } from '../../../icons';

import { Button } from '..';
import { COLORS } from '../Button/constants';

import './styles.css';

const getClassName = (className) =>
	classNames('ButtonWithArrow', className);

const ButtonWithArrow = ({ className, children, ...rest }) => (
	<Button
		primary
		color={COLORS.WHITE}
		className={getClassName(className)}
		{...rest}
	>
		<span>
			{children}
			<ArrowIcon />
		</span>
	</Button>
);

ButtonWithArrow.propTypes = {
	children: PropTypes.any,
	className: PropTypes.string,
};

export default ButtonWithArrow;
