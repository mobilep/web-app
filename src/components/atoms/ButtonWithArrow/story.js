import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs/react';

import ButtonWithArrow from './ButtonWithArrow';

storiesOf('1. Atoms / ButtonWithArrow', module)
	.add('ButtonWithArrow', () => <ButtonWithArrow text={text('Text', 'Sign In')} />);
