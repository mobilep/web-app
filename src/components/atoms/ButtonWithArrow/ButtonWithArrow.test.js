import React from 'react';
import { shallow } from 'enzyme';

import ButtonWithArrow from './ButtonWithArrow';

describe('Atom - ButtonWithArrow', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<ButtonWithArrow>
				Label
			</ButtonWithArrow>
		);
		expect(wrapper).toMatchSnapshot();
	});

});

