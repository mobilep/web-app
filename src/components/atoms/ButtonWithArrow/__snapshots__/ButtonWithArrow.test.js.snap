// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - ButtonWithArrow Should render 1`] = `
<Button
  className="ButtonWithArrow"
  color="white"
  primary={true}
>
  <span>
    Label
    <ArrowIcon />
  </span>
</Button>
`;
